src/executionengine/lljit.jl

```diff
@@ -0,0 +1,82 @@
+@checked struct LLJITBuilder
+    ref::API.LLVMOrcLLJITBuilderRef
+end
+Base.unsafe_convert(::Type{API.LLVMOrcLLJITBuilderRef}, builder::LLJITBuilder) = builder.ref
+
+@checked mutable struct LLJIT
+    ref::API.LLVMOrcLLJITRef
+end
+Base.unsafe_convert(::Type{API.LLVMOrcLLJITRef}, lljit::LLJIT) = lljit.ref
+
+function LLJITBuilder()
+    ref = API.LLVMOrcCreateLLJITBuilder()
+    LLJITBuilder(ref)
+end
+
+function dispose(builder::LLJITBuilder)
+    API.LLVMOrcDisposeLLJITBuilder(builder)
+end
+
+function targetmachinebuilder!(builder::LLJITBuilder, tmb::TargetMachineBuilder)
+    API.LLVMOrcLLJITBuilderSetJITTargetMachineBuilder(builder, tmb)
+end
+
+function linkinglayercreator!(builder::LLJITBuilder, callback, ctx)
+    API.LLVMOrcLLJITBuilderSetObjectLinkingLayerCreator(builder, callback, ctx)
+end
+
+"""
+    LLJIT(::LLJITBuilder)
+
+Creates a LLJIT stack based on the provided builder.
+
+!!! note
+    Takes ownership of the provided builder.
+"""
+function LLJIT(builder::LLJITBuilder)
+    ref = Ref{API.LLVMOrcLLJITRef}()
+    @check API.LLVMOrcCreateLLJIT(ref, builder)
+    LLJIT(ref[])
+end
+
+function LLJIT(; tm::Union{Nothing, TargetMachine} = nothing)
+    builder = LLJITBuilder()
+    if tm === nothing
+        tmb = TargetMachineBuilder()
+    else
+        tmb = TargetMachineBuilder(tm)
+    end
+    targetmachinebuilder!(builder, tmb)
+    LLJIT(builder)
+end
+
+function dispose(lljit::LLJIT)
+    API.LLVMOrcDisposeLLJIT(lljit)
+end
+
+function LLJIT(f::Core.Function, builder::LLJITBuilder)
```

Comment:
Maybe unify both these by just forwarding `args...` (and `kwargs...` if you want to keep the `tm` a keyword argument).